CRM/Core/Menu.php

```diff
@@ -112,8 +112,10 @@ public static function &xmlItems($fetchFromXML = FALSE) {
    * @throws Exception
    */
   public static function read($name, &$menu) {
+    libxml_disable_entity_loader(FALSE);
     $xml = simplexml_load_file($name);
     self::readXML($xml, $menu);
+    libxml_disable_entity_loader(TRUE);
```

Comment:
The general pattern for doing this safely as well-behaved member of a complex system is:

```php
$backupValueThatIDontMakeAnyAssumptionsAbout = readTheGlobalValue():
setTheGlobalValue($theValueThatILike);
try {
   doTheMainThing();
}
finally {
  setTheGlobalValue($backupValueThatIDontMakeAnyAssumptionsAbout);
}
```

On a case-by-cases basis, there might be a prettier way (e.g. maybe depending on the contract of the specific global-thing or  maybe using `CRM_Utils_AutoClean::swap()` or maybe using `CRM_Utils_GlobalStack`).